docs/msbuild/customize-your-build.md

```diff
@@ -86,6 +98,51 @@ A summary of MSBuild's general approach is as follows:
 
 Or more simply: the first *Directory.Build.props* which doesn't import anything, is where MSBuild stops.
 
-## See Also  
- [MSBuild Concepts](../msbuild/msbuild-concepts.md)   
- [MSBuild Reference](../msbuild/msbuild-reference.md)   
+## MSBuildProjectExtensionsPath
+
+By default, `Microsoft.Common.props` imports `$(MSBuildProjectExtensionsPath)$(MSBuildProjectFile).*.props` and `Microsoft.Common.targets` imports `$(MSBuildProjectExtensionsPath)$(MSBuildProjectFile).*.targets`. The default value of `MSBuildProjectExtensionsPath` is `$(BaseIntermediateOutputPath)`, `obj/`. This is the mechanism that NuGet uses to refer to build logic delivered with packages--at restore time, it creates `{project}.nuget.g.props` files that refer to the package contents.
+
+This extensibility mechanism can be disabled by setting the property `ImportProjectExtensionProps` to `false` before importing `Microsoft.Common.props`.
+
+## MSBuildExtensionsPath and MSBuildUserExtensionsPath
+
+> [!WARNING]
+> Using these extension mechanisms makes it harder to get repeatable builds across machines. Prefer configuration that can be checked into your source control system and shared among all developers of your codebase.
+
+By convention, many core build logic files import
+
+```
+$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\{TargetFileName}\ImportBefore\*.targets
+```
+
+before their contents, and
+
+```
+$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\{TargetFileName}\ImportAfter\*.targets
+```
+
+afterward. This allows installed SDKs to augment the build logic of common project types.
+
+The same directory structure is searched in `$(MSBuildUserExtensionsPath)`, which is the per-user folder `%LOCALAPPDATA%\Microsoft\MSBuild`. Files placed in that folder will be imported for all builds of the corresponding project type run under that user's credentials.
```

Comment:
Consider listing the MSBuild properties that can be used to override this and when they can be set in the build.

Reply:
😳 I looked for and **completely** failed to see those properties.

Reply:
There are also off switches for the installed wildcard directories; I am deliberately _not_ mentioning them here because they're _very_ likely to break projects. For example, PackageReferences only work because NuGet installs itself to `Microsoft.Common.props\ImportBefore\Microsoft.NuGet.ImportBefore.props`.